instagram-app/src/components/CommentSection/CommentContainer.js

```diff
@@ -0,0 +1,62 @@
+import React, { Fragment, Component } from "react";
+import { shape, arrayOf, string } from "prop-types";
+import Comment from "./Comment";
+import CommentInputField from "./CommentInputField";
+import moment from "moment";
+
+class CommentContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      comments: props.comments,
+      commentInput: ""
+    };
+  }
+
+  addComment = event => {
+    event.preventDefault();
+    const username = localStorage.getItem("user");
+    const newComment = {
+      username: username,
+      text: this.state.commentInput,
+      timestamp: moment().format("M/D/YYYY, h:mm:ss a")
+    };
+
+    //  Spread the existing array adding the new comment
+    this.setState({
+      comments: [...this.state.comments, newComment],
+      commentInput: ""
+    });
+  };
+
+  handleChanges = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  };
+  render() {
+    return (
+      <Fragment>
+        {this.state.comments.map((e, i) => {
+          return <Comment key={i} comment={e} />;

```

Comment:
You never want to use the index from an array method as a key. Usually you'll have an id on your data and you can use that